modules/widgets/social-media-icons.php

```diff
@@ -325,6 +325,18 @@ public function remove_username( $val ) {
  * @return void
  */
 function wpcom_social_media_icons_widget_load_widget() {
-	register_widget( 'wpcom_social_media_icons_widget' );
+	$transient = 'wpcom_social_media_icons_widget::is_active';
+	$has_widget = get_transient( $transient );
+
+	if ( false === $has_widget ) {
+		$is_active_widget = is_active_widget( false, false, 'wpcom_social_media_icons_widget', false );
+		$has_widget       = (int) ! empty( $is_active_widget );
+		set_transient( $transient, $has_widget, 1 * HOUR_IN_SECONDS );
+	}
+
+	// [DEPRECATION]: Only register widget if active widget exists already
+	if ( 1 === $has_widget ) {
```

Comment:
I think there's some weird type handling going on - I was only able to get deprecated widgets shown on my site after I removed the strict type check from this if, and made it `if( $has_widget) `. Maybe `get_transient` or `set_transient` do something funny here?
Anyway, seems to be working with this one nitpick.